### PR TITLE
devops: Add GHA workflow for triggering semgrep-perf-benchmarks Argo Workflow

### PR DESCRIPTION
**Pull request by minusworld**
This differs from the `semgrep-comparison` workflow.

[This previous PR](https://github.com/semgrep/semgrep/pull/9239) builds and pushes a container image containing the `perf/` directory for easily running these perf tests on `pr-NNNN` builds. This PR adds a GHA for triggering the container image in Argo Workflows.

The purpose of this is to eventually offload the benchmarks from GHA onto Argo, which will allow parallelizing the benchmarks for faster runtimes.

### Test plan

Manually trigger the Argo WF URL.

```
✗ curl -X POST https://argoworkflows-dev2.corp.r2c.dev/api/v1/events/security-research/semgrep-perf-benchmarks -H "..." -d '{"repository": "semgrep/WebGoat", "sha": "78d5bf174b43296f79d33ea26398f39cf681bffc", "image_tag": "pr-9255-performance-tests"}'
{}%
```

image

### CODE REVIEW

**Review by apt-itude — APPROVED** [review]

**Comment by aryx** [comment]
@minusworld can you add @IagoAbal for all those PRs as at least one person in PA need to understand those workflows I think.

File: .github/workflows/trigger-semgrep-perf-benchmarks-argo.jsonnet [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  jobs: {
+    'setup-docker-tag': util.setup_docker_tag(),
+    'get-sha': util.get_sha(),
+    'trigger-semgrep-perf-benchmarks-argo': util.trigger_argo_workflow(
+      "https://argoworkflows-dev2.corp.r2c.dev/api/v1/events/security-research/semgrep-perf-benchmarks",
+      [
+        {"name": "image_tag", "value": "${{ needs.setup-docker-tag.outputs.docker-tag }}"},
+      ],
+    ),
```
Comment (emjin) [inline .github/workflows/trigger-semgrep-perf-benchmarks-argo.jsonnet:28]: One thing I was curious about: does `trigger_argo_workflow` assume the presence of `$SHA` and `$REPOSITORY`?

**Review by emjin — APPROVED** [review]
Makes sense to me. It does seem a little fragile---I left a comment asking about some assumptions, but we're also assuming that our docker tag naming stays the same. I'm not super concerned, just noted that.

Is the next step to make this workflow run on PR?

**Comment by aryx** [comment]
@IagoAbal @minusworld do we still need this? Should we merge it?

**Comment by aryx** [comment]
closing to cleanup the opened PRs but feel free to reopen if still relevant

**Pull request closed without merging**